android/quest/src/main/java/org/smartregister/fhircore/quest/ui/appsetting/AppSettingActivity.kt

```diff
@@ -67,18 +70,35 @@ class AppSettingActivity : AppCompatActivity() {
     val existingAppId =
       sharedPreferencesHelper.read(SharedPreferenceKey.APP_ID.name, null)?.trimEnd()
 
+
+
     // If app exists load the configs otherwise fetch from the server
     if (!existingAppId.isNullOrEmpty()) {
```

Comment:
The only check we need to update will be this line. The function `loadConfigurations` will write the app id from the `local.properties` file to shared preferences.

```suggestion
    if (!existingAppId.isNullOrEmpty() || !BuildConfig.APP_ID.isNullOrEmpty()) {
```

The rest of the implementation need not to change.

Reply:
The `ConfigurationRegistry.loadConfigurations` will handle the error message if the application ID  is not found in the assets directory for debug variants.

Reply:
If we do that, the app would just freeze in the case of an non-debug app id e.g "wdf". This is because `ConfigurationRegistry.loadConfigurations()` would not be trying to load from assets and this piece of code in the same function does not throw anything when a composition does not exist i.e `?.` returns null


`else {
      fhirEngine.searchCompositionByIdentifier(parsedAppId)?.run {
        populateConfigurationsMap(context, this, false, parsedAppId, configsLoadedCallback)
      }
`

Reply:
I see. You are right, I missed the point that non-debug app configurations are downloaded from the server. My suggestion assumed the user had previously launched the app. Lemme re-check and revert.

Reply:
Borrowing from your implementation we can include the scenario when APP_ID Is provided in the local.properties in the if statement. Then use `fetchConfiguration` since it uses `loadConfiguration` internally. 

```kotlin
 // If app exists load the configs otherwise fetch from the server
    if (!existingAppId.isNullOrEmpty()) {
      appSettingViewModel.run {
        onApplicationIdChanged(existingAppId)
        loadConfigurations(appSettingActivity)
      }
    } else if (BuildConfig.OPENSRP_APP_ID.isNotEmpty()) {
      appSettingViewModel.run {
        onApplicationIdChanged(BuildConfig.OPENSRP_APP_ID)
        fetchConfigurations(appSettingActivity)
      }
    } else {
      setContent {
        AppTheme {
          val appId by appSettingViewModel.appId.observeAsState("")
          val showProgressBar by appSettingViewModel.showProgressBar.observeAsState(false)
          val error by appSettingViewModel.error.observeAsState("")
          AppSettingScreen(
            appId = appId,
            onAppIdChanged = appSettingViewModel::onApplicationIdChanged,
            fetchConfiguration = appSettingViewModel::fetchConfigurations,
            showProgressBar = showProgressBar,
            error = error,
          )
        }
      }
    }
```

The conditions should be in the order above. Should resolve the issue with app freezing.

Reply:
Can you also rename the config property to `OPENSRP_APP_ID`. There is an existing `APPLICATION_ID` configuration provided via Android gradle config, using `APP_ID` will be confusing.

Reply:
When using the OPENSRP_APP_ID from `local.properties`  we should not display the Setting screen at all to the user. It's okay for the application to force close with an error message if the configuration is missing.

Reply:
> When using the OPENSRP_APP_ID from `local.properties` we should not display the Setting screen at all to the user. It's okay for the application to force close with an error message if the configuration is missing.

Thank you for clearing this up!